swarmcd/stack.go

```diff
@@ -78,17 +98,31 @@ func (swarmStack *swarmStack) updateStack() (revision string, err error) {
 	log.Debug("rotating configs and secrets...")
 	err = swarmStack.rotateConfigsAndSecrets(stackContents)
 	if err != nil {
-		return
+		return "", fmt.Errorf("failed to rotate configs and secrets for %s stack: %w", swarmStack.name, err)
 	}
 
 	log.Debug("writing stack to file...")
-	err = swarmStack.writeStack(stackContents)
+	writtenBytes, err := swarmStack.writeStack(stackContents)
 	if err != nil {
-		return
+		return "", fmt.Errorf("failed to write stack to file for %s stack: %w", swarmStack.name, err)
+	}
+
+	if !swarmStack.shouldDeploy(writtenBytes, deployedStackHash, revision, lastRevision) {
+		return lastRevision, nil
```

Comment:
should we update the db to the current revision anyway?

That way the next check would be quicker and exit at line 65

Reply:
I absolutely see your point. However, we were thinking about this, and we were arguing whether it is better to see in the UI, which revision contained the last change, and don't display just the most recent one.

But I am not sure, which option is better here.

Reply:
I circumvented this question by introducing an enriched `stackMetadata` type, which stores the `repoRevision` and the `deployedStack` revision.